Remove "tags" entry from URL

josephsnyder opened the pull request:
Replace the SOLR entry in the query URL for the tags entry.
Switch from nested ifs to a switch statement to make expansion easier.

josephsnyder committed:
Remove "tags" entry from URL

Replace the SOLR entry in the query URL for the tags entry.
Switch from nested ifs to a switch statement to make expansion easier.

BetsyMcPhail commented:
I'm getting a syntax error for the tag 's':

`<span style="color:red;">Error: midas.journal.search?NaN&query=text-journal.enable:true AND (text-journal.tag(s):s OR name:(tags:s) OR description:(tags:s) OR ngram_search:(tags:s)) AND ( text-journal.community:1 )&level=&secondQuery=text-journal.enable:true AND (text-journal.tag(s):s OR name:(tags:s) OR description:(tags:s) OR ngram_search:(tags:s)) AND ( text-journal.community:1 )&useSession=true failed: Syntax error in query </span><br>`

josephsnyder committed:
Add colon to value replace in search string generation

Add a ":" to the search string generation when the plain text values
are replaced with the SOLR index string. This will prevent the value
from being placed into the SOLR index string.
